hy/compiler.py

```diff
@@ -1737,10 +1737,35 @@ def compile_require(self, expression):
         "unimport" it after we've completed `thing' so that we don't pollute
         other envs.
         """
-        expression.pop(0)
-        for entry in expression:
-            __import__(entry)  # Import it fo' them macros.
-            require(entry, self.module_name)
+        for entry in expression[1:]:
+            if isinstance(entry, HySymbol):
+                # e.g., (require foo)
```

Comment:
REALLY minor nitpick, but this would be a little easier to read without the `e.g.`, since I think it's obvious that it's an example.